src/devices/cpu/tlcs900/tmp94c241.cpp

```diff
@@ -0,0 +1,1798 @@
+	m_port_latch[PORT_D] = 0x00;
+	m_port_latch[PORT_E] = 0x00;
+	m_port_latch[PORT_F] = 0x00;
+	m_port_latch[PORT_H] = 0x00;
+	m_port_latch[PORT_Z] = 0x00;
+
+	m_port_function[PORT_0] = 0x01;
+	m_port_function[PORT_1] = 0x01;
+	m_port_function[PORT_2] = 0x01;
+	m_port_function[PORT_3] = 0x01;
+	m_port_function[PORT_4] = 0xff;
+	m_port_function[PORT_5] = 0xff;
+	m_port_function[PORT_6] = 0xff;
+	m_port_function[PORT_7] = 0x01;
+	m_port_function[PORT_8] = 0x00;
+	m_port_function[PORT_A] = 0x00;
+	m_port_function[PORT_B] = 0x00;
+	m_port_function[PORT_C] = 0x00;
+	m_port_function[PORT_D] = 0x00;
+	m_port_function[PORT_E] = 0x00;
+	m_port_function[PORT_F] = 0x00;
+	m_port_function[PORT_H] = 0x00;
+	m_port_function[PORT_Z] = 0x00;
+
+	std::fill_n(&m_port_control[0], NUM_PORTS, 0x00);
+	m_t8run = 0x00;
+	m_trdc = 0x00;
+	std::fill_n(&m_tmod[0], 8, 0x00);
+	std::fill_n(&m_ffcr[0], 6, 0x00);
+	std::fill_n(&m_timer[0], 4, 0x00);
+	std::fill_n(&m_timer16[0], 4, 0x00);
+	m_watchdog_mode = 0x80;
+	for( int i = 0; i < 2; i++ )
+	{
+		m_serial_control[i] &= 0x80;
+		m_serial_mode[i] &= 0x80;
+		m_baud_rate[i] = 0x00;
+	}
+	m_od_enable = 0x00;
+	m_ad_mode1 = 0x00;
+	m_ad_mode2 = 0x00;
+	std::fill_n(&m_int_reg[0], 18, 0x00);
+	m_iimc = 0x00;
+	std::fill_n(&m_dma_vector[0], 4, 0x00);
+	m_block_cs[0] = 0x0000;
+	m_block_cs[1] = 0x0000;
+	m_block_cs[2] = 0x1000; //FIXME!
+	m_block_cs[3] = 0x0000;
+	m_block_cs[4] = 0x0000;
+	m_block_cs[5] = 0x0000;
+	m_external_cs = 0x0000;
+	std::fill_n(&m_mem_start_reg[0], 6, 0xff);
+	std::fill_n(&m_mem_start_mask[0], 6, 0xff);
+	std::fill_n(&m_dram_refresh[0], 2, 0x00);
+	std::fill_n(&m_dram_access[0], 2, 0x80);
+	m_da_drive = 0x00;
+}
+
+uint8_t tmp94c241_device::inte_r(offs_t offset)
+{
+	return m_int_reg[offset];
+}
+
+void tmp94c241_device::inte_w(offs_t offset, uint8_t data)
+{
+	if ( data & 0x80 )
+		data = ( data & 0x7f ) | ( m_int_reg[offset] & 0x80 );
+	if ( data & 0x08 )
+		data = ( data & 0xf7 ) | ( m_int_reg[offset] & 0x08 );
+
+	m_int_reg[offset] = data;
+	m_check_irqs = 1;
+}
+
+uint8_t tmp94c241_device::intnmwdt_r(offs_t offset)
+{
+	return m_int_reg[INTNMWDT];
+}
+
+void tmp94c241_device::intnmwdt_w(offs_t offset, uint8_t data)
+{
+	if ( data & 0x80 )
+		data = ( data & 0x7f ) | ( m_int_reg[INTNMWDT] & 0x80 );
+	if ( data & 0x08 )
+		data = ( data & 0xf7 ) | ( m_int_reg[INTNMWDT] & 0x08 );
+
+	m_int_reg[INTNMWDT] = data;
+	m_check_irqs = 1;
+}
+
+void tmp94c241_device::iimc_w(uint8_t data)
+{
+	m_iimc = data;
+	m_check_irqs = 1;
+}
+
+void tmp94c241_device::intclr_w(uint8_t data)
+{
+	for( int i = 0; i < NUM_MASKABLE_IRQS; i++ )
+	{
+		if ( data == tmp94c241_irq_vector_map[i].dma_start_vector )
+		{
+			// clear interrupt request
+			m_int_reg[tmp94c241_irq_vector_map[i].reg] &= ~ tmp94c241_irq_vector_map[i].iff;
+			return;
+		}
+	}
+}
+
+void tmp94c241_device::dmav_w(offs_t offset, uint8_t data)
+{
+	m_dma_vector[offset] = data;
+}
+
+void tmp94c241_device::b0cs_w(offs_t offset, uint8_t data)
```

Comment:
Why don't you use 16-bits handlers and COMBINE_DATA ?

Reply:
I don't know how that works. Can you help me?

Reply:
void tmp94c241_device::b0cs_w(offs_t offset, uint16_t data, uint16_t mem_mask)
{
    COMBINE_DATA(&m_block_cs[0]);
}

Reply:
Also implemented this suggestion. Thanks.